### DISCUSSION TITLE
2.11.0-rc.1

### DISCUSSION THREAD

**Discussion by ruibaby** [halo-dev/halo#4925] [General]
## ⚠️ 注意

此版本由于升级了 SpringBoot 的版本并依赖了新的 H2 Database 版本，经过我们测试无法正常从旧版本 Halo 升级。所以如果你当前正在使用 H2 作为数据库，可能需要使用系统的[备份恢复](https://docs.halo.run/user-guide/backup)功能进行重新安装。

此外，不建议在生产环境使用 H2 作为数据库。

## 新特性

- 文章预览支持模拟不同设备尺寸。 by @Thematrixofink in <https://github.com/halo-dev/halo/pull/4846>
- 支持个人中心。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4857>
- 新增用户邮箱验证机制。 by @guqing in <https://github.com/halo-dev/halo/pull/4878>
- 支持为自定义的角色配置登录之后默认跳转位置。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4918>
- 新增用户站内消息删除功能。 by @guqing in <https://github.com/halo-dev/halo/pull/4906>
- 用户注册新增电子邮箱字段。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4899>
- 支持在概览页面显示已启动插件和已激活的主题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4897>
- 使用 ibm-semeru-runtimes:open-17-jre 作为基础镜像构建 Halo 镜像，以获得更优的内存消耗。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4896>

## 功能优化

- 提供允许同步更改插件运行状态的 API。 by @guqing in <https://github.com/halo-dev/halo/pull/4745>
- 优化 Console 端插件捆绑资源的请求，防止在操作插件状态之后无法及时更新资源。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4804>
- 修改发布文章的等待时间以防止因数据库执行延迟较高导致的错误提示。 by @guqing in <https://github.com/halo-dev/halo/pull/4810>
- 添加 Generator 元数据标识。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4821>
- 异步加载文件上传组件，优化 Console 加载速度。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4875>
- 优化 Console 端表单元素样式。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4893>
- 为默认编辑器中 heading 类型元素生成有意义的 id。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4909>
- 优化初始化预设插件的流程。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4901>

## 问题修复

- 修复筛选未发布文章时结果不正确的问题。 by @guqing in <https://github.com/halo-dev/halo/pull/4736>
- 修复 Console 端评论状态筛选无法显示筛选结果的问题。 by @zjy4fun in <https://github.com/halo-dev/halo/pull/4743>
- 修复部分场景下插件详情的扩展 tab 不能正常显示的问题。 by @longjuan in <https://github.com/halo-dev/halo/pull/4746>
- 解决文章设置中可能出现空白元数据的问题。 by @LIlGG in <https://github.com/halo-dev/halo/pull/4772>
- 修复无法正常调用评论列表 Finder API 的问题。 by @Roozenlz in <https://github.com/halo-dev/halo/pull/4773>
- 修复文章可能因为没有发布时间导致发布失败的问题。 by @guqing in <https://github.com/halo-dev/halo/pull/4775>
- 解决无用户管理权限的用户无法上传头像的问题。 by @LIlGG in <https://github.com/halo-dev/halo/pull/4777>
- 修复因部分角色导致无法正常创建个人令牌的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4800>
- 优化编辑器性能并解决切换页面所造成的卡死现象。 by @LIlGG in <https://github.com/halo-dev/halo/pull/4805>
- 解决删除元数据后仍旧提示 key 不能重复的问题。 by @LIlGG in <https://github.com/halo-dev/halo/pull/4815>
- 修复在部分 Windows 开发者无法正常构建 Halo 的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4844>
- 修复手动刷新搜索索引会将私有文章加入索引的问题。 by @AirboZH in <https://github.com/halo-dev/halo/pull/4882>
- 修复因错误构建插件的路由可能导致 OOM 的问题。 by @guqing in <https://github.com/halo-dev/halo/pull/4890>
- 修复在 Card 组件中使用下拉组件无法完整显示的问题。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4922>

## 依赖更新

- 升级默认编辑器版本。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4924>
	- 解决行内块元素宽度设置不准确的问题。 by @LIlGG in <https://github.com/halo-sigs/richtext-editor/pull/71>
- 升级 Gradle 至 8.4。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4848>
- 升级 Lombok Gradle 插件至 8.4。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4849>
- [Action Required] 升级 Spring Boot 至 3.2.0-RC2。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4850>
- 升级 SpringDoc 至 2.2.1-SNAPSHOT 以解决部分 API 文档无法正常使用的问题。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4854>
- 升级 FormKit 的版本至 1.3.0。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4913>
- 升级 Spring Boot 依赖至 3.2.0。 by @JohnNiang in <https://github.com/halo-dev/halo/pull/4904>

## 开发者相关

- 重构 Repeater 组件，使其支持条件判断。 by @LIlGG in <https://github.com/halo-dev/halo/pull/4702>
- contributorFinder 支持返回用户元数据、注册信息、用户权限等。 by @AirboZH in <https://github.com/halo-dev/halo/pull/4705>
- 暴露 Actuator 指标端点以便监控服务状态。 by @guqing in <https://github.com/halo-dev/halo/pull/4898>
- 添加 `@halo-dev/ui-plugin-bundler-kit` 包，用于向插件提供统一的构建配置。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4916>
- 添加 `<StickyBlock />` 组件，用于将元素固定在顶部或者底部，并为主题 / 插件 / 系统设置的底部保存按钮区域做了适配。 by @ruibaby in <https://github.com/halo-dev/halo/pull/4919>

<hr /><em>This discussion was created from the release <a href='https://github.com/halo-dev/halo/releases/tag/v2.11.0-rc.1'>2.11.0-rc.1</a>.</em>

**Comment by AirboZH** [comment]
内存占用减少了超级多 🚀 COOL!!!
